### PR TITLE
Fb ux assay test migrate

### PR DESCRIPTION
**Pull request by labkey-chrisj**
This is the test migration of assay tests, to use the new react-based assayDesigner

This feature branch has related branches in other modules:
platform, commonAssay, biologics <-- these 3 have tests that depend on changes in this branch

### CODE REVIEW

File: src/org/labkey/test/tests/SampleSetParentColumnTest.java [hunk 1]
```diff
@@ -692,42 +691,32 @@ public void testAliasNameConflictsWithFieldName()
...
+        clickFolder(SUB_FOLDER_NAME);
+        DomainFormPanel domainFormPanel = sampleHelper.goToEditSampleSetFields(SAMPLE_SET_NAME);
+        domainFormPanel.addField(GOOD_PARENT_NAME);
+        clickButton("Save", 0);
 
-        fieldProperties.addField();
+        errorMsgLocator = Locator.tagWithClass("div", "alert-danger");
+        waitForElement(errorMsgLocator);
+        String errorMsgExpectedTxt = "'" + GOOD_PARENT_NAME + "' is a reserved field name in '" + SAMPLE_SET_NAME + "'.";
+        Assert.assertTrue("Error message not as expected.", errorMsgLocator.findElement(getDriver()).getText().contains(errorMsgExpectedTxt));
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleSetParentColumnTest.java]: Use hamcrest assertion to generate better error messages.
```suggestion
        Assert.assertThat("Error message", errorMsgLocator.findElement(getDriver()).getText(), CoreMatchers.containsString(errorMsgExpectedTxt));
```

File: src/org/labkey/test/tests/SampleSetTest.java [hunk 2]
```diff
@@ -122,12 +121,15 @@ private void doSetup()
...
+        goToProjectHome();
         projectMenu().navigateToProject(PROJECT_NAME);
         portalHelper.addWebPart("Sample Sets");
 
+        goToProjectHome();
         projectMenu().navigateToFolder(PROJECT_NAME, FOLDER_NAME);
         portalHelper.addWebPart("Sample Sets");
 
+        goToProjectHome();
         projectMenu().navigateToFolder(PROJECT_NAME, LINEAGE_FOLDER);
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleSetTest.java]: These `goToProjectHome` calls shouldn't be necessary. There's a bug in the project menu helper if they are.

Comment (cnathe) [reply src/org/labkey/test/tests/SampleSetTest.java]: Correct. I have removed them. I had added them when testing something locally and forgot to remove. Done.

File: src/org/labkey/test/tests/SampleSetParentColumnTest.java (hunk 1 above)
Comment (cnathe) [reply src/org/labkey/test/tests/SampleSetParentColumnTest.java]: Done.

File: src/org/labkey/test/pages/user/ShowUsersPage.java [hunk 3]
```diff
@@ -49,12 +44,6 @@ public DataRegionTable getUsersTable()
         return elementCache().usersTable;
     }
 
-    public PropertiesEditor clickChangeUserProperties()
-    {
-        waitAndClickAndWait(Locator.linkWithSpan( "Change User Properties"));
-        return new PropertiesEditor.PropertiesEditorFinder(getDriver()).withTitle("Field Properties").waitFor();
-    }
-
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/pages/user/ShowUsersPage.java]: Yes. I added it back and updated to the correct return type.

File: src/org/labkey/test/WebDriverWrapper.java [hunk 4]
```diff
@@ -821,6 +822,15 @@ public ShowUsersPage goToSiteUsers()
...
     }
 
+    public DomainFormPanel goToChangeUserProperties()
+    {
+        enableUxDomainDesigner();
+        goToSiteUsers();
+        clickButton("Change User Properties", defaultWaitForPage);
+        disableUxDomainDesigner();
+        return new DomainFormPanel.DomainFormPanelFinder(getDriver()).withTitle("Users").find();
+    }
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/WebDriverWrapper.java]: Done.

File: src/org/labkey/test/pages/experiment/CreateSampleSetPage.java [hunk 5]
```diff
@@ -90,10 +89,10 @@ public String getParentAliasSelectText(int index)
         return new Select(elementCache().parentAliasSelect(index)).getFirstSelectedOption().getText();
     }
 
-    public EditDomainPage clickCreate()
+    public DomainFormPanel clickCreate(boolean createFailureExpected)
     {
         clickAndWait(elementCache().createButton);
-        return new EditDomainPage(getDriver());
+        return createFailureExpected ? null : new DomainFormPanel.DomainFormPanelFinder(getDriver()).waitFor();
     }
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/pages/experiment/CreateSampleSetPage.java]: Done.

File: src/org/labkey/test/WebDriverWrapper.java [hunk 6]
```diff
@@ -864,6 +874,16 @@ public ManageStudyPage goToManageStudy()
...
 
+    public DomainFormPanel goToStudyEditAdditionalProperties(boolean goThroughManageStudy)
+    {
+        enableUxDomainDesigner();
+        if (goThroughManageStudy)
+            goToManageStudy();
+        waitAndClickAndWait(Locator.linkWithText("Edit Additional Properties"));
+        disableUxDomainDesigner();
+        return new DomainFormPanel.DomainFormPanelFinder(getDriver()).withTitle("Study").find();
+    }
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/WebDriverWrapper.java]: Done.

**Pull request merged**